src/index.js

```diff
@@ -425,27 +432,67 @@ const registerLoginRoute = (keycloak) => {
     });
 };
 
+const registerDropSessionMethod = (keycloak) => {
+    keycloak.server.method('dropSession', (sessionId) => {
+        const cache = keycloak.server._core.caches.get(keycloak.config.cacheName);
+        if (cache) {
+            return cache.client.drop({id: sessionId, segment: '!yar'})
+        }
+        throw new Error(`Cannot find the "${keycloak.config.cacheName}" cache for drop procedure.`);
+    })
+};
+
+const registerBackChannelLogoutRoute = (keycloak) => {
+    keycloak.server.route({
+        path: keycloak.config.backChannelLogoutUrl,
+        method: 'POST',
+        handler: async (request, reply) => {
+            keycloak.server.log(['debug', 'keycloak'], 'Back-channel logout');
+            const logoutToken = new Token(request.payload);
```

Comment:
I'm afraid creating a `Token` object is not good enough here.
An action token must be validated (i.e. via `GrantManager.validateToken` method). Otherwise anyone can revoke access to any session.
Additionally, it must be checked that `logoutToken.content.action === 'LOGOUT'`